src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -6251,19 +6255,19 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
 #endif
                 llvm::Value* llvm_size = arr_descr->get_array_size(target, nullptr, 4);
                 target = llvm_utils->CreateLoad2(target_el_type->getPointerTo(), arr_descr->get_pointer_to_data(target));
-                value = llvm_utils->create_gep(value, 0);
+                value = llvm_utils->create_gep2(value_type_, value, 0);
                 llvm::DataLayout data_layout(module->getDataLayout());
                 uint64_t data_size = data_layout.getTypeAllocSize(value_el_type);
                 llvm_size = builder->CreateMul(llvm_size,
                     llvm::ConstantInt::get(context, llvm::APInt(32, data_size)));
                 builder->CreateMemCpy(target, llvm::MaybeAlign(), value, llvm::MaybeAlign(), llvm_size);
             } else if( is_target_data_only_array || is_value_data_only_array ) {
                 if( is_value_fixed_sized_array ) {
-                    value = llvm_utils->create_gep(value, 0);
+                    value = llvm_utils->create_gep2(value_el_type, value, 0);
```

Comment:
```suggestion
value = llvm_utils->create_gep2(value_type_, value, 0);
```

I think you have to use `value_type_` here.

Reply:
Can you once rebase and check on your device if the issue persists or it is due to the CI.

Reply:
The main problem is i have ran the test and kep breakpoints on all my changes control never reached any of those points.

Reply:
Okay. I am checking it.